impeller/renderer/pipeline.h

```diff
@@ -87,8 +87,73 @@ PipelineFuture<ComputePipelineDescriptor> CreatePipelineFuture(
     const Context& context,
     std::optional<ComputePipelineDescriptor> desc);
 
+/// This is a classed use to check that the input slots of fragment shaders
+/// match the output slots of the vertex shaders.
+/// It's not used at runtime.
+template <typename VertexShaderT, typename FragmentShaderT>
+class ShaderBindingComparator {
+ public:
+  static constexpr bool CompileTimeStrEqual(const char* str1,
+                                            const char* str2) {
+    return *str1 == *str2 &&
+           (*str1 == '\0' || CompileTimeStrEqual(str1 + 1, str2 + 1));
+  }
+
+  static constexpr bool Compare() {
```

Comment:
Perhaps name this "AreStagesCompatible"? With a docstring that says something like, "checks if the for vertex stage outputs are compatible with the fragment stage. If not, attempting to create a pipline at runtime will lead to link errors."

Reply:
I renamed it Check and added a docstring.  FYI this won't lead to runtime errors but will create compile time errors when building the engine.

Reply:
Oh wait, i see, the link errors are what we are trying to avoid?  I'll add a comment.

Reply:
done